src/components/Datatable/BatchModule/ElementCounter/ElementCounter.tsx

```diff
@@ -30,22 +44,49 @@ const getCounterContent = (totalLength, selectedLength = 0) => {
   return <LightText>No data</LightText>;
 };
 
-const ElementCounter: React.FC<ElementCounterProps> = ({
-  totalLength,
-  selectedLength = 0,
-}) => {
-  const content = getCounterContent(totalLength, selectedLength);
+const ElementCounter: React.FC = () => {
+  const {
+    totalLength = 0,
+    selectedLength,
+    hasExclusionLogic,
+    setHasExclusionLogic,
+    hasSelection,
+  } = useDatatable();
+  const computedSelectedLength = hasExclusionLogic
+    ? totalLength - selectedLength
+    : selectedLength;
+
+  const content = getCounterContent(totalLength, computedSelectedLength);
+  const handleSelectClick = useCallback(
+    (state) => {
+      setHasExclusionLogic(state);
+    },
+    [setHasExclusionLogic],
```

Comment:
For those useCallbacks, is correct assumption that the setter functions never update and will be used just their first instance?

Reply:
yep it's internal DT state which isn't customizable and not exposed